app/models/bot/alegre.rb

```diff
@@ -96,32 +96,40 @@ def add_relationships(pm, pm_ids)
     r.save!
   end
 
-  def get_image_similarities(pm)
-    return if pm.report_type != 'uploadedimage' or CONFIG['vframe_host'].blank?
-
-    require 'net/http/post/multipart'
-
-    # Send image to VFRAME to get matches.
-    url = URI.parse(CONFIG['vframe_host'] + '/api/v1/match')
-    response = { 'results' => [] }
-    context = self.get_context(pm).to_json
-    Net::HTTP.start(url.host, url.port, :use_ssl => url.scheme == 'https') do |http|
-      req = Net::HTTP::Post::Multipart.new(url, {
-        'url' => pm.media.file.file.public_url,
-        'context' => context,
-        'filter' => { project_id: pm.project.id }.to_json,
-        'threshold' => 1,
-        'limit' => 1
-      })
-
-      begin
-        response = JSON.parse(http.request(req).body)
-      rescue StandardError => e
-        Rails.logger.error("[Alegre Bot] Bad response from VFRAME: #{e.message}")
-        self.class.notify_error(e, { bot_id: self.id, vframe_url: url, context: context }, RequestStore[:request] )
-      end
-    end
-    pm_ids = response.dig('results')&.collect{|r| r.dig('context', 'project_media_id')}
+  def self.get_image_similarities(pm)
+    return if pm.report_type != 'uploadedimage'
+
+    # Query for similar images.
+    similar = self.request_api('get', '/image/similarity', {
+      url: pm.media.file.file.public_url,
+      context: self.get_context(pm),
```

Comment:
If we send the full `context`, with both `team_id` and `project_media_id`, it means that on the Alegre side it will try to match the whole context and thus always return only the same project media, no?